### PR TITLE
 Oracle Compatibility: Support for instr function

### PR DESCRIPTION
**Pull request by NotHimmel**
## Summary by CodeRabbit

* **New Features**
  * Added Oracle-compatible INSTR string-search functions with overloads to support different argument combinations (string, substring, start position, occurrence).

* **Tests**
  * Added comprehensive SQL tests for INSTR covering NULLs, varchar/text inputs, varied start positions and occurrence counts, and multiple example phrases.

### CODE REVIEW

File: contrib/ivorysql_ora/expected/ora_character_datatype_functions.out [hunk 1]
```diff
@@ -3128,6 +3128,77 @@ select decode(null, null, 2018, 2, 3) from test_decode; --expect:all
 (4 rows)
 
 drop table test_decode;
+
+/*
+ * instr
+ */
+
+SELECT INSTR('OBJECT-ORIENTED PROGRAMMING', 'O') FROM DUAL;
```
Comment (c2main) [inline contrib/ivorysql_ora/expected/ora_character_datatype_functions.out:3154]: Is it worth adding test for NULL input ? The functions are defined STRICT at SQL level, and the C part has no protection from NULL input.

File: contrib/ivorysql_ora/sql/ora_character_datatype_functions.sql [hunk 2]
```diff
@@ -1198,6 +1198,22 @@ select decode(null, null, 2018, 2, 3) from test_decode; --expect:all
 
 drop table test_decode;
 
+/*
+ * instr
+ */
+
+SELECT INSTR('OBJECT-ORIENTED PROGRAMMING', 'O') FROM DUAL;
```
Comment (c2main) [inline contrib/ivorysql_ora/sql/ora_character_datatype_functions.sql:1211]: there is  `varchar2` and `text` input support, test should cover both using explicit cast.

File: contrib/ivorysql_ora/src/builtin_functions/character_datatype_functions.c [hunk 3]
```diff
@@ -1722,6 +1725,45 @@ appendUTF16Escape(StringInfoData* outputString, uint16_t codePoint) {
     appendStringInfoString(outputString, buffer);
 }
 
+/****************************************************************
+ * instr
+ *
+ * Syntax:
+ *   instr (string VARCHAR, substring VARCHAR)
```
Comment (c2main) [inline contrib/ivorysql_ora/src/builtin_functions/character_datatype_functions.c]: it's `TEXT` just after this comment, not `VARCHAR`)

File: contrib/ivorysql_ora/expected/ora_character_datatype_functions.out (hunk 1 above)
Comment (NotHimmel) [reply contrib/ivorysql_ora/expected/ora_character_datatype_functions.out:3154]: Thanks! You're right. 
I'll add the following test cases to ensure proper NULL handling:

```sql
-- NULL input tests
SELECT INSTR(NULL, 'O') FROM DUAL;      -- Expected: NULL
SELECT INSTR('TEST', NULL) FROM DUAL;   -- Expected: NULL  
SELECT INSTR(NULL, NULL) FROM DUAL;     -- Expected: NULL
```

File: contrib/ivorysql_ora/sql/ora_character_datatype_functions.sql (hunk 2 above)
Comment (NotHimmel) [reply contrib/ivorysql_ora/sql/ora_character_datatype_functions.sql:1211]: I'll add tests for both VARCHAR2 and TEXT support:

```sql
-- VARCHAR2 
SELECT INSTR('OBJECT-ORIENTED PROGRAMMING', 'O') FROM DUAL;

-- TEXT
SELECT INSTR(CAST('OBJECT-ORIENTED PROGRAMMING' AS TEXT), CAST('O' AS TEXT)) FROM DUAL;
SELECT INSTR('OBJECT-ORIENTED PROGRAMMING'::TEXT, 'O') FROM DUAL;
```

File: contrib/ivorysql_ora/src/builtin_functions/character_datatype_functions.c (hunk 3 above)
Comment (NotHimmel) [reply contrib/ivorysql_ora/src/builtin_functions/character_datatype_functions.c]: Fixed in next commit.

**Review by yuanyl630 — APPROVED** [review]
